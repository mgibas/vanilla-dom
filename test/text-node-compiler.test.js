const compiler = require('../src/text-node-compiler')
let textNodeUpdateCompiler = require('../src/text-node-update-compiler')
textNodeUpdateCompiler.compile = jest.fn()
textNodeUpdateCompiler.compile.mockReturnValue('UPDATE_COMPILER_PART')

describe('text-node-compiler', () => {
  let options = {state:'st'}

  describe('on compile', () => {
    it('simple text node', () => {
      expect(compiler.compile('parent', {data: 'node data'}, options)).toMatchSnapshot();
    })
    it('text node with expression', () => {
      expect(compiler.compile('parent', {data: 'Helloo {{st.fullname}}!'}, options)).toMatchSnapshot();
    })
  })
})
