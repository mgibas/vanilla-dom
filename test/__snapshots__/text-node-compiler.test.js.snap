// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-node-compiler on compile simple text node 1`] = `
"(() => {
  let textNode = document.createTextNode(\`node data\`);
  parent.appendChild(textNode);
})();"
`;

exports[`text-node-compiler on compile text node with expression 1`] = `
"(() => {
  let textNode = document.createTextNode(\`Helloo \${ (() => {
    try {
      let val = st.fullname;
      return val !== null && val !== undefined ? val : '';
    } catch (err) {
      return '';
    }
  })() }!\`);
  parent.appendChild(textNode);
  UPDATE_COMPILER_PART;
})();"
`;
